docs/controllers/gc.md

```diff
@@ -0,0 +1,112 @@
+
+    You should use owner references on generated child objects that have a clear owner object within Kubernetes.
+
+To successfully use owner references you need to:
+
+1. insert the reference on any object you are creating from your reconciler
+2. mark the children as watchable through owner [[relations]]
+
+### Owner Reference Example
+In the [configmapgen_controller example](https://github.com/kube-rs/kube/blob/main/examples/configmapgen_controller.rs), the controller creates a `ConfigMap` from a contrieved `ConfigMapGenerator` custom resource ([cmg crd](https://github.com/kube-rs/kube/blob/main/examples/configmapgen_controller_crd.yaml)). The [example's reconciler](https://github.com/kube-rs/kube/blob/83368df52a4845e06edbb9b4b3246c3807bb711a/examples/configmapgen_controller.rs#L37-L73) for the `ConfigMapGenerator` objects insert the `owner_reference` into the generated `ConfigMap`:
+
+```rust
+let oref = generator.controller_owner_ref(&()).unwrap();
+let cm = ConfigMap {
+    metadata: ObjectMeta {
+        name: generator.metadata.name.clone(),
+        owner_references: Some(vec![oref]),
+        ..ObjectMeta::default()
+    },
+    data: Some(contents),
+    ..Default::default()
+};
+```
+
+using [`Resource::controller_owner_ref`](https://docs.rs/kube/latest/kube/trait.Resource.html#method.controller_owner_ref). It then [marks](https://github.com/kube-rs/kube/blob/83368df52a4845e06edbb9b4b3246c3807bb711a/examples/configmapgen_controller.rs#L108C11-L109) `ConfigMap` as a dependent type to watch for via [[relations]]:
+
+```rust
+Controller::new(cmgs, watcher::Config::default())
+    .owns(cms, watcher::Config::default())
+```
+
+## Finalizers
+A finalizer is a marker on a root object that indicates that a controller will perform cleanup if it is ever deleted. Kubernetes will block the object from being deleted until the controller completes this action, and subsequently takes itself of the finalizer list.
+
+This is explained in more detail in [Kubernetes.io :: Finalizers](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/).
+
+!!! note "Finalizers mark the need for controller cleanup"
+
+    You should mark objects with a finalizer if it needs external cleanup to run in the event it is deleted.
+
+The main way to use finalizers with controllers is to define a unique finalizer name (many controllers can finalize an object) and make the [finalizer] helper manage it. The finalizer helper is desingned to be used within a reconciler and split the work depending on the state we are in:
+
+- have a deletion happened, and we need to cleanup? we are in the `Event::Cleanup` arm
+- no deletion has been recorded? we are in the normal `Event::Apply` arm
+
+
+!!! warning "Finalizer prevent objects from being deleted"
+
+    If your controller is down, or for some other reason is not able to complete the cleanup (or remove the finalizer from the object), the object will not be deleted.
```

Comment:
Maybe this is an interpretation issue, but I could read this as "if controller is down then deletes will be ignored", rather than "if controller is down then deletes will be delayed until the controller is back".

Reply:
Yeah, good point. Reworded these bits in https://github.com/kube-rs/website/pull/43/commits/dc2c63f38dbf85a3013da04e17f35e0e21511b25